Check is Airship Flying

Gimbal-Myk opened the pull request:
We can't expect users to have called UAirship.takeOff() in time so we need to check if Airship is taking off or not before sending airship events.

Gimbal-Myk committed:
add check for UAirship.isFlying() before sending Airhsip events
maven publish changes

Gimbal-Myk committed:
implemented queue for creating Airship Events.

Gimbal-Myk committed:
went for an iterative approach instead of recursive.

Gimbal-Myk committed:
pr suggs

Gimbal-Myk committed:
pr suggs

kimvsparrow commented on airship-adapter/src/main/java/com/gimbal/airship/AirshipAdapter.java:
```diff
@@ -34,7 +34,9 @@
 import java.util.HashMap;
+import java.util.LinkedList;
 import java.util.List;
+import java.util.NoSuchElementException;
```
is this exception used?

Gimbal-Myk committed:
pr suggs

Gimbal-Myk committed:
pr suggs

kimvsparrow committed:
Publish to Maven Central staging repo

Setup Gradle Nexus Publish Plugin.  Add artifact signature.

Update Airship dependency on the way.

kimvsparrow commented on airship-adapter/src/main/java/com/gimbal/airship/AirshipAdapter.java:
```diff
@@ -519,4 +539,14 @@ protected void onPostExecute(Boolean result) {
     }
+}
+
+class CachedVisit {
```
last thing I think: make this a `private static` inner class.

Gimbal-Myk committed:
pr suggs

Gimbal-Myk committed:
Merge remote-tracking branch 'origin/check-isflying' into check-isflying
